applications/mne_analyze/plugins/dataloader/dataloader.cpp

```diff
@@ -217,6 +252,56 @@ void DataLoader::onSaveFilePressed()
                                                     tr("Fiff file(*.fif *.fiff)"));
 
     QFileInfo fileInfo(sFilePath);
-    m_pSelectedModel->saveToFile(sFilePath);
+
+    triggerLoadingStart("Saving file...");
+
+    connect(&m_FutureWatcher, &QFutureWatcher<void>::finished,
+            this, &DataLoader::saveFileEnd, Qt::UniqueConnection);
+
+    m_Future = QtConcurrent::run(this, &DataLoader::saveFile, sFilePath);
```

Comment:
This is potentially dangerous since the saveFile function is not Thread safe. For example it would crash if the model would be removed by the user while the saveFile function is still running in a different thread. Same for loading the file.